Disher/View Controllers/RecipesViewController.m

```diff
@@ -87,29 +84,63 @@ - (void) queryMealDB:(NSString *) name completionHandler:(void(^)(NSArray *retur
-                                              handler:^(UIAlertAction * action) {
-                   [self viewDidLoad];
-               }];
-
-               [alert addAction:defaultAction];
-               [self presentViewController:alert animated: YES completion: nil];
+           if (error != nil) {//TODO: add error message
            }
            else {
                NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
-               completionHandler(dataDictionary[@"meals"]);
+               if (dataDictionary[@"meals"] == [NSNull null] ) {
+                   NSLog(@"null detected form mealdb");
+               }
+               else {
+                   completionHandler(dataDictionary[@"meals"]);
+               }
 
            }
     }];
     [task resume];
 }
 
+- (void) querySearchSpoonacular:(NSString *) name completionHandler:(void(^)(NSArray *returnedMeals))completionHandler {
+    //Use API Key in Keys.plist file
+    NSString *path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
+    NSString *key = [dict objectForKey: @"spoon_key"];
+    NSString *apiKeyArg = [NSString stringWithFormat:@"&apiKey=%@", key];
+    NSString *queryURL = [NSString stringWithFormat:@"https://api.spoonacular.com/recipes/complexSearch?query=\"%@\"%@", name, apiKeyArg];
+    queryURL = [queryURL stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
+    
+    NSURL *url = [NSURL URLWithString:queryURL];
+    NSURLRequest *request = [NSURLRequest requestWithURL:url cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+           if (error != nil) { //TODO: Add error message
+           }
+           else {
+               NSDictionary *dataDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
+               if (![(NSArray *)dataDictionary[@"results"] count]) {
+                   NSLog(@"null detected from spoonacular");
+               }
+               else {
+                   completionHandler(dataDictionary[@"results"]);
+               }
+           }
+    }];
+    [task resume];
+}
+
+- (void) queryAPIs:(NSString *) input completionHandler:(void(^)(void))completionHandler {
+    [self querySearchSpoonacular:input completionHandler:^(NSArray *returnedMeals) {
+        self.spoonResults = returnedMeals;
+        completionHandler();
```

Comment:
you will end up reloading the table twice; here and in line 138. can you think of a way to prevent this?